capstone/src/main/java/com/google/step/servlets/PageViewServlet.java

```diff
@@ -0,0 +1,132 @@
+     *                 getAllPageViews() -> List<RestaurantPageViews> (as Json list)
+     * @throws IOException can be thrown by the writer for the response.
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String restaurantKey;
+        if (request.getParameter("restaurantKey") != null) {
+            restaurantKey = request.getParameter("restaurantKey");
+        } else {
+            restaurantKey = null;
+        }
+
+        int year = 0;
+        if (request.getParameter("year") != null) {
+            year = Integer.parseInt(request.getParameter("year"));
+        }
+
+        String json;
+        if (restaurantKey == null && year == 0) {
+            // getAllPageViews()
+            List<RestaurantPageViews> restaurantPageViewsList = metricsClient.getAllPageViews();
+            json = convertRestaurantPageViewListToJsonUsingGson(restaurantPageViewsList);
+        } else if (year == 0 && restaurantKey != null) {
+            // getCurrentPageViews()
+            WeeklyPageView weeklyPageView = metricsClient.getCurrentPageViews(restaurantKey);
+            json = convertWeeklyPageViewToJsonUsingGson(weeklyPageView);
+
+        } else if (year != 0 && restaurantKey != null) {
+            // getYearRestaurantPageViews
+            List<WeeklyPageView> weeklyPageViewList = metricsClient.getYearRestaurantPageViews(year, restaurantKey);
+            json = convertWeeklyPageViewListToJsonUsingGson(weeklyPageViewList);
+        } else {
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+
+    /**
+     *
+     * @param request should contain the <String, restaurantKey> and <String, restaurantName> as
+     *                parameters. A SC_FORBIDDEN status can be sent to the frontend if one of these
+     *                parameters is not present.
+     * @param response no response.
+     * @throws IOException can be thrown by the response.
+     */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String restaurantKey = request.getParameter("restaurantKey");
+        String restaurantName = request.getParameter("restaurantName");
+
+        if (restaurantKey == null || restaurantName == null) {
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+
+        metricsClient.putPageView(restaurantName, restaurantKey);
+    }
+
+    /**
+     * Converts to Json using Gson.
+     * @param weeklyPageView Map to convert to Gson.
+     * @return String in Json format.
+     */
+    private static String convertWeeklyPageViewToJsonUsingGson(WeeklyPageView weeklyPageView) {
+        Gson gson = new Gson();
```

Comment:
Could you instead have a `Gson` object as a member variable instead of repeated initializing new ones?

Reply:
Same comment as above.